### PR TITLE
BFB-416: Platform Fix

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-416

**Description**
- Currently the platform doesn't reset and clean up for each world.
- Solution:
- Added `EnvironmentManager` to handle current and future environmental objects on the map. 
- Added Checkbox for environments on the map

**Screenshots / GIFs**

### CODE REVIEW

File: Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs [hunk 1]
```diff
@@ -52,20 +54,29 @@ protected WeaponSceneManager WeaponSceneManager
 
         protected virtual void CleanUp()
         {
-            Debug.Log("Cleaning Up");
+            if (_environmentManager != null)
+            {
+                _environmentManager.CleanUp();
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs:59]: If this is here, should we remove the same cleanup in the `WaveSandboxWorldManager` clean up?
https://github.com/PerigonGames/BForBoss/blob/c8de3e278ea85bed897070bf8fd74ace6d263eb9/Assets/_BForBoss/_Core/Scripts/Managers/WaveSandboxWorldManager.cs#L22

File: Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public override void Reset()
             _healthbar.Reset();
         }
 
-        protected override void CleanUp()
+        public override void CleanUp()
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:62]: Nitpick: looks like `TriggerHitAnimation` and `LifeCycleFinished` are private and protected respectively and should below the public methods

File: Assets/_BForBoss/_Entities/Scripts/LifeCycleBehaviour.cs [hunk 3]
```diff
@@ -67,7 +67,7 @@ protected virtual void LifeCycleDamageTaken()
             }
         }
 
-        protected virtual void CleanUp()
+        public virtual void CleanUp()
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Entities/Scripts/LifeCycleBehaviour.cs:70]: Nitpick: public should be above protected

File: Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs:59]: i thought i removed it already. lemme ccheck

File: Assets/_BForBoss/_Entities/Scripts/LifeCycleBehaviour.cs (hunk 3 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Entities/Scripts/LifeCycleBehaviour.cs]: Fixed

File: Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:50]: Fixed

File: Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs:59]: Fixed

**Review by Gursaanj — APPROVED** [review]

**Pull request merged**